kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DomainMetadataSuite.scala

```diff
@@ -540,4 +541,75 @@ class DomainMetadataSuite extends DeltaTableWriteSuiteBase with ParquetSuiteBase
       }
     }
   }
+
+  test(
+    "RowTrackingMetadataDomain is serializable and deserializable"
+  ) {
+    withTempDirAndEngine((tablePath, engine) => {
+      // Create a RowTrackingMetadataDomain
+      val rowTrackingMetadataDomain = new RowTrackingMetadataDomain(10)
+
+      // Generate a DomainMetadata action from it.
+      val dm = rowTrackingMetadataDomain.toDomainMetadata
+      // The configuration string should be a JSON serialization of the rowTrackingMetadataDomain
+      assert(dm.getDomain === rowTrackingMetadataDomain.getDomainName)
+      assert(dm.getConfiguration === """{"rowIdHighWaterMark":10}""")
+
+      // Verify the deserialization from DomainMetadata action into concrete domain object
+      val deserializedDomain = RowTrackingMetadataDomain.fromJsonConfiguration(dm.getConfiguration)
+      assert(deserializedDomain === rowTrackingMetadataDomain)
+
+      // Verify the domainMetadata can be committed and read back
+      createTableWithDomainMetadataSupported(engine, tablePath)
+      // Commit the domain metadata and verify
+      commitDomainMetadataAndVerify(
+        engine,
+        tablePath,
+        domainMetadatas = Seq(dm),
+        expectedValue = Map(rowTrackingMetadataDomain.getDomainName -> dm)
+      )
+
+      // Read the domain metadata back from the table snapshot
+      val table = Table.forPath(engine, tablePath)
+      val snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      val rowTrackingMetadataDomainFromSnapshot =
+        RowTrackingMetadataDomain.fromSnapshot(snapshot).get
+
+      // Verify the domain metadata read back from the snapshot
+      assert(rowTrackingMetadataDomain === rowTrackingMetadataDomainFromSnapshot)
+    })
+  }
+
+  test(
+    "JsonMetadataDomain deserialization fails if there are missing/extra fields"
+  ) {
+    // Create a RowTrackingMetadataDomain, which is a concrete JsonMetadataDomain implementation
+    val rowTrackingMetadataDomain = new RowTrackingMetadataDomain(10)
+
+    // Serialize it to a JSON configuration
+    val config = rowTrackingMetadataDomain.toJsonConfiguration
+    assert(config === """{"rowIdHighWaterMark":10}""")
+
+    // Deserialization should fail if the configuration string has extra fields
+    val configExtraField = """{"rowIdHighWaterMark":10, "extraField": "extra"}"""
```

Comment:
This will cause issues by the way. We seem to have a bug in Delta Spark where we accidentally also include the domain name in the serialized json:
```json
{"domainMetadata":{"domain":"delta.rowTracking","configuration":"{\"rowIdHighWaterMark\":0,\"domainName\":\"delta.rowTracking\"}","removed":false}}
```
(We forgot to `@JsonIgnore` this field and never noticed...)

Reply:
Interestingly, I found this from Jackson's doc

> When ignoring the whole property, the default behavior if encountering such property in input is to ignore it without exception; but if there is a JsonAnySetter it will be called instead. Either way, no exception will be thrown.

So just have an extra field `domainName` should be fine, since it has `@JsonIgnore` in Delta Kernel. I've added integration tests to test this (write `RowTrackingMetadataDomain` with Kernel and read with Delta-Spark, and vice versa).

Reply:
That's great to hear, that saves us from a giant headache. Could you add a small test that explicitly checks this case? That way we can make sure that we remain backwards compatible in the future (after we've fixed Delta Spark to no longer include the domain name).